### PR TITLE
feat(domain): Add domain record data source

### PR DESCRIPTION
**Pull request by jerjako**

### CODE REVIEW

File: scaleway/data_source_domain_record_test.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+					data scaleway_domain_record test {
+						dns_zone  = "${scaleway_domain_record.main.dns_zone}"
+						record_id = "${scaleway_domain_record.main.id}"
+					}
+
+					data scaleway_domain_record test2 {
+						dns_zone = "${scaleway_domain_record.main.dns_zone}"
+						name     = "${scaleway_domain_record.main.name}"
+						type     = "${scaleway_domain_record.main.type}"
+						data     = "${scaleway_domain_record.main.data}"
```
Comment (Monitob) [inline scaleway/data_source_domain_record_test.go:37]: Can we retrieve the ttl information?

File: scaleway/data_source_domain_record.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+)
+
+func dataSourceScalewayDomainRecord() *schema.Resource {
+	// Generate datasource schema from resource
+	dsSchema := datasourceSchemaFromResourceSchema(resourceScalewayDomainRecord().Schema)
+
+	// Set 'Optional' schema elements
+	addOptionalFieldsToSchema(dsSchema, "dns_zone", "name", "type", "data")
+
+	dsSchema["name"].ConflictsWith = []string{"record_id"}
```
Comment (Monitob) [inline scaleway/data_source_domain_record.go:20]: Please add the doc in /docs/data-sources/domain.md

File: scaleway/data_source_domain_record_test.go (hunk 1 above)
Comment (Monitob) [reply scaleway/data_source_domain_record_test.go:37]: It's possible to retrieve the geoip data?

**Review by Monitob — CHANGES_REQUESTED** [review]

File: scaleway/data_source_domain_record_test.go (hunk 1 above)
Comment (jerjako) [reply scaleway/data_source_domain_record_test.go:38]: done, I found a bug doing that. Name and type was not exported in the `resourceScalewayDomainRecordRead`

**Review by Monitob — APPROVED** [review]

**Pull request merged**